src/v/coproc/pacemaker.cc

```diff
@@ -68,17 +68,20 @@ pacemaker::pacemaker(unresolved_address addr, ss::sharded<storage::api>& api)
 ss::future<> pacemaker::start() {
     co_await ss::recursive_touch_directory(offsets_snapshot_path().string());
     auto ncc = co_await recover_offsets(_offs.snap, _shared_res.api.log_mgr());
-    _ntps = std::move(ncc);
+    std::exchange(_ntps, std::move(ncc));
```

Comment:
it doesn't look like you need exchange here. you could `_ntps = co_await ...` or the removed line with std::move(ncc).